pkg/auth/providers/azure/azure_provider.go

```diff
@@ -400,6 +415,28 @@ func formAzureRedirectURL(config map[string]interface{}) string {
 	)
 }
 
+func extractAnnotations(config map[string]interface{}) map[string]string {
+	annotations := make(map[string]string)
+	metadata, ok := config["metadata"].(map[string]interface{})
+	if !ok {
+		logrus.Infof("Failed to decode the %q field of the auth config.", "metadata")
```

Comment:
Nit:

I don't think that it's helpful to use a format here given that the entire statement is a constant. There's arguments here from a readability perspective (why are we formatting something when the arg is a constant) and from an efficiency perspective (instead of just printing out the line, now logrus has to parse the format string, insert the argument, render the statement, and log it out). 

Marked as nit because it's pretty minor.

Reply:
I agree, but I did this because of I wanted to give the same form to these TWO logging statements. If there was only one, I would not use the argument.

Reply:
I get that the log statements have the same form, but since the format isn't a variable, we don't get any benefits of having them the same. 

I.E. - if I want to change this, I still need to make a change in two places rather than one. So really they are only the same because the string in two places are the same (just like it would be if the entire log statement was one statement).